here-naksha-lib-psql/src/commonTest/kotlin/naksha/psql/ReadFeaturesByMetadataTest.kt

```diff
@@ -0,0 +1,412 @@
+                    TupleColumn.type(),
+                    StringOp.EQUALS,
+                    inputFeature.type
+                )
+            )
+        );
+
+        // Then:
+        assertEquals(1, featuresCreatedInFrame.features.size)
+        assertEquals(inputFeature.id, featuresCreatedInFrame.features[0]!!.id)
+    }
+
+    @Test
+    fun shouldReadFeatureByUpdatedInTimeFrame(){
+        // Given:
+        val inputFeature = ProxyFeatureGenerator.generateRandomFeature().apply {
+            type = "type_for_updated_at_frame_test"
+        }
+
+        // And:
+        val insertedFeatureMeta = insertFeatureAndGetMeta(inputFeature)
+
+        // And:
+        val featuresUpdatedInFrame = executeMetaQuery(
+            MetaAnd(
+                MetaQuery(
+                    TupleColumn.updatedAt(),
+                    DoubleOp.GTE,
+                    insertedFeatureMeta.updatedAt
+                ),
+                MetaQuery(
+                    TupleColumn.updatedAt(),
+                    DoubleOp.LTE,
+                    insertedFeatureMeta.updatedAt + 100
+                ),
+                MetaQuery(
+                    TupleColumn.type(),
+                    StringOp.EQUALS,
+                    inputFeature.type
+                )
+            )
+        );
+
+        // Then:
+        assertEquals(1, featuresUpdatedInFrame.features.size)
+        assertEquals(inputFeature.id, featuresUpdatedInFrame.features[0]!!.id)
+    }
+
+    @Test
+    fun shouldReadFeatureByAuthorTs(){
+        // Given:
+        val inputFeature = ProxyFeatureGenerator.generateRandomFeature()
+
+        // And:
+        val insertedFeatureMeta = insertFeatureAndGetMeta(inputFeature)
+
+        // And:
+        val featuresByAuthorTs = executeMetaQuery(MetaQuery(
+            TupleColumn.authorTs(),
+            DoubleOp.EQ,
+            insertedFeatureMeta.authorTs
+        ))
+
+        // Then:
+        assertEquals(1, featuresByAuthorTs.features.size)
+        assertEquals(inputFeature.id, featuresByAuthorTs.features[0]!!.id)
+
+    }
+
+    @Test
+    fun shouldReadFeatureByMetadataAlternative() {
+        // Given
+        val appId = "some_app"
+        val featuresToCreate = ProxyFeatureGenerator.generateRandomFeatures(count = 10)
+
+        // When
+        insertFeatures(featuresToCreate, SessionOptions(appId = appId))
+
+        // And: execute
+        val featuresByAppIdAndAuthor = executeRead(ReadFeatures().apply {
+            collectionIds += collection!!.id
+            query.metadata = MetaOr(
+                MetaQuery(TupleColumn.author(), StringOp.EQUALS, "this_is_totally_off"),
+                MetaQuery(TupleColumn.appId(), StringOp.STARTS_WITH, appId.substring(0, 2))
+            )
+        }).features
+
+        // Then:
+        assertEquals(10, featuresByAppIdAndAuthor.size)
+        assertTrue(featuresByAppIdAndAuthor.map { it!!.id }
+            .containsAll(featuresToCreate.map { it.id }))
+    }
+
```

Comment:
Maybe also a test for when the value type does not match the column type? Example `authorId` is 100.

Reply:
I think Paweł had similar comment to your Property search recently and we decided that in that case, the query is legal but won't return anything - let's be consistent, I personally like that we don't maintain such checks

Reply:
I think it might still be worth it to include a test here. Indeed we agreed that the features will simply be filtered out and not returned, but without tests it might simply work in unexpected way (every feature is a match and returned, or maybe the value is casted if possible and used in the query,...). Anyhow treat this as a nice to have recommendation (I also didn't add this kind of test for PropertyFilter like you said :) ).